### ISSUE TITLE
Clarification on 3.1.2 Language of Parts

### ISSUE THREAD

**Issue by goodwitch** [w3c/wcag#297]
So, I need clarification on [3.1.2 Language of Parts ](https://www.w3.org/TR/WCAG20/#meaning-other-lang-id)

Is it a failure if 
- (an html web site has alternative language links to versions of the page in other languages (e.g., Deutsch, Français, Nederlands, Castellano, etc.)) AND
- (the text of each link is the name of the language, in that language) AND
- (the language of each link is indicated via a lang attribute) AND
- (the text of each language link does not have a language attribute to indicate the change in language?

Why am I asking this question?  Because I've been calling this as a failure...but one of my colleagues just pointed out that the normative part of this SC says "passage or phrase", so they tell me, that since these language links are single words, they are not required to be marked.  

My brain says...sometimes a phrase can be a single word.  And I've found evidence of this in linguistics.  

If you agree that this is a failure, would it be worthwhile to add this as a failure technique for 3.1.2 in WCAG 2.0?

Thanks for considering this.

**Comment by Ryladog** [comment]
My interpretation has always been that single word links to other language pages that appear in the secondary language is in fact a failure.

 

​​​​​* katie *

 

Katie Haritos-Shea 
Principal ICT Accessibility Architect (WCAG/Section 508/ADA/AODA)

 

Cell: 703-371-5545 |  <mailto:ryladog@gmail.com> ryladog@gmail.com | Oakton, VA |  <http://www.linkedin.com/in/katieharitosshea/> LinkedIn Profile | Office: 703-371-5545 |  <https://twitter.com/Ryladog> @ryladog

NOTE: The content of this email should be construed to always be an expression of my own personal independent opinion, unless I identify that I am speaking on behalf of Knowbility, as their AC Rep at the W3C - and - that my personal email never expresses the opinion of my employer, Deque Systems.

**Comment by srinivasuchakravarthula** [comment]
I agree. 

Regards,
Srinivasu Chakravarthula
+91-9900810881
Sent on my phone. Excuse typos, if any.

**Comment by kerstinp** [comment]
@goodwitch, all,

this example is mentioned in the Understanding of SC 3.1.2 (Example 2). So it seems that a single word can be seen as a "phrase". But because there is no definition of "phrase" in the glossary it's not very lucky when just the informative understanding document indicates this. That's why I filed an issue for WCAG 2.1 that a "definition of phrase in the glossary is needed": https://github.com/w3c/wcag21/issues/287.

In some german texts about this aspect it is said that of course it is a benefit and a must when correct lang-attributes for longer passages of texts have been used. But it is also written that the benefit for single words is lower and can also be disturbing when many single words with lang-attribute are present because of a short pause in the reading flow of JAWS when switching in the other language and back. (Of course the SC says that not every single word of another language must be defined). This pause in the reading flow seems to be nowadays in comparison with ten years ago shorter but seems to be still there.

I think one aspect is: As far as I know english speaking people tend to pronounce words from other languages with english pronounciation (for example "kindergarten") and the amount of words from other languages seems to be not so high than in other languages. In comparison with english the amount of words from other languages in german texts for example is in general much higher and we tend to pronounce for example english words in english.

I think that a failure technique with some examples would be very helpful to bring in more clarification in  SC 3.1.2. And I also think that the Working Group should seek for feedback on this issue / future failure techniques especially from screenreader user from countries where english is not native language.

**Comment by Ryladog** [comment]
Kerstin,

Thanks for you comments. The Understanding document of 2.0 and Techniques
documents can be/arehi updated regularly, so tbis identification of perhaps
an issue with 3.1.2 may be addressed at any time. It does not have to be
attached to WCAG 2.1. So it may be something we can look at and clarify
after 2.1 launches.

Katie Haritos-Shea
703-371-5545

On Jul 25, 2017 1:55 AM, "Kerstin Probiesch" <notifications@github.com>
wrote:

@goodwitch <https://github.com/goodwitch>, all,

this example is mentioned in the Understanding of SC 3.1.2 (Example 2). So
it seems that a single word can be seen as a "phrase". But because there is
no definition of "phrase" in the glossary it's not very lucky when just the
informative understanding document indicates this. That's why I filed an
issue for WCAG 2.1 that a "definition of phrase in the glossary is needed":
w3c/wcag21#287 <https://github.com/w3c/wcag21/issues/287>.

In some german texts about this aspect it is said that of course it is a
benefit and a must when correct lang-attributes for longer passages of
texts have been used. But it is also written that the benefit for single
words is lower and can also be disturbing when many single words with
lang-attribute are present because of a short pause in the reading flow of
JAWS when switching in the other language and back. (Of course the SC says
that not every single word of another language must be defined). This pause
in the reading flow seems to be nowadays in comparison with ten years ago
shorter but seems to be still there.

I think one aspect is: As far as I know english speaking people tend to
pronounce words from other languages with english pronounciation (for
example "kindergarten") and the amount of words from other languages seems
to be not so high than in other languages. In comparison with english the
amount of words from other languages in german texts for example is in
general much higher and we tend to pronounce for example english words in
english.

I think that a failure technique with some examples would be very helpful
to bring in more clarification in SC 3.1.2. And I also think that the
Working Group should seek for feedback on this issue / future failure
techniques especially from screenreader user from countries where english
is not native language.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/w3c/wcag/issues/297#issuecomment-317637269>, or mute
the thread

**Comment by alastc** [comment]
Hi everyone, I'm going through old issues...
For the scenario where you have a link to another page with the name of the language in the language (e.g. "Français"), the understanding doc has that example.

I wasn't worried by the use of "phrase", that could be a word.

I did wonder if you could consider Français a "proper name" for France? In which case you could exempt that... However I think that scenario of language links is fairly important to get right, for people who need that language to hear it pronounced properly. 

Overall, does anyone want to sign up for creating a technique/failure for this? 
If not, I suggest we defer or simply close this one.

**Comment by patrickhlauke** [comment]
> I did wonder if you could consider Français a "proper name" for France

well, a proper name for "French" (in French). but i'd say no to the larger question. And that language links that just have a single word (the language, in that language) need `lang` or it's a failure.

```
<ul>
  <li><a href="..." lang="de">Deutsch</a></li>
  <li><a href="..." lang="it">Italiano</a></li>
  <li><a href="..." lang="fr">Français</a></li>
  ...
  <li><a href="..." lang="zh-hant">繁體中文</a></li>
</ul>
```

**Comment by alastc** [comment]
I think the nub of this issue is clarifying that a phrase can be a word. 

Maybe change the first line of the intent in the [understanding doc](https://www.w3.org/WAI/WCAG21/Understanding/language-of-parts.html) to:
> The intent of this Success Criterion is to ensure that user agents can correctly present words, phrases or passages written in multiple languages.

**Comment by cstrobbe** [comment]
> My brain says...sometimes a phrase can be a single word. And I've found evidence of this in linguistics.

Yes, a single word can count as a phrase. To quote the [definition of phrase from Wiktionary](https://en.wiktionary.org/wiki/phrase#Noun):

> (*grammar*) A word or, more commonly, a group of words that functions as a single unit in the [syntax](https://en.wiktionary.org/wiki/syntax) of a sentence, usually consisting of a [head](https://en.wiktionary.org/wiki/head), or central word, and elaborating words.

When I studied linguistics at university, one of the things I learnt was that there is no definition for "word" that works across languages. For this reason, the "ambiguity" of the term "phrase" can be seen as an advantage.

So I agree with other commenters that something like `<li><a href="..." lang="fr">français</a></li>` requires a `lang` attribute. (Names of languages are written in lowercase in French.)

**Comment by CharlesBelov** [comment]
See also  3.1.2 Language of parts - can single words without lang mark-up be considered a PASS? #1174

**Comment by ZackaryJacobthereal** [comment]
Null

**Comment by awkawk** [comment]
Also related to: https://github.com/w3c/wcag21/issues/287
